cmd/mdatagen/testdata/config_gen/expected_golang_output/batch

```diff
@@ -0,0 +1,63 @@
+// Code generated by github.com/atombender/go-jsonschema, DO NOT EDIT.
+
+package batchprocessor
+
+import "encoding/json"
+
+// Configuration parameters for the batch processor.
+type Config struct {
+	// MetadataCardinalityLimit corresponds to the JSON schema field
+	// "metadata_cardinality_limit".
+	MetadataCardinalityLimit int `json:"metadata_cardinality_limit,omitempty" yaml:"metadata_cardinality_limit,omitempty" mapstructure:"metadata_cardinality_limit,omitempty"`
+
+	// MetadataKeys corresponds to the JSON schema field "metadata_keys".
+	MetadataKeys []string `json:"metadata_keys,omitempty" yaml:"metadata_keys,omitempty" mapstructure:"metadata_keys,omitempty"`
+
+	// SendBatchMaxSize corresponds to the JSON schema field "send_batch_max_size".
+	SendBatchMaxSize int `json:"send_batch_max_size,omitempty" yaml:"send_batch_max_size,omitempty" mapstructure:"send_batch_max_size,omitempty"`
+
+	// SendBatchSize corresponds to the JSON schema field "send_batch_size".
+	SendBatchSize int `json:"send_batch_size,omitempty" yaml:"send_batch_size,omitempty" mapstructure:"send_batch_size,omitempty"`
+
+	// Timeout corresponds to the JSON schema field "timeout".
+	Timeout string `json:"timeout,omitempty" yaml:"timeout,omitempty" mapstructure:"timeout,omitempty"`
```

Comment:
Is it very important that durations are represented by `time.Duration` instead of `string`? 

I'm also not sure if we could upstream such a feature in [go-jsonschema](https://github.com/omissis/go-jsonschema). Maybe someone more experienced with the project can opine? Otherwise I could raise an issue upstream.

Reply:
The [earlier PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27003/files#diff-8ba245fc89fb661229e773df7f5a0e17e1c269f8b29f3d01b37a75270035bebbR16) was able to get the translation to output `time.Duration` due to custom changes to the `go-jsonschema` fork. I'll try to apply the same changes.

Reply:
I raised a [PR](https://github.com/omissis/go-jsonschema/pull/257) upstream so that it supports `time.Duration`. I consider this a pre-requisite for the OTel PR to be merged.